components/expenses/ExpenseAttachedFiles.js

```diff
@@ -21,16 +21,16 @@ const ExpenseAttachedFiles = ({ files, onRemove, showInvoice, collective, expens
                 onClick={downloadInvoice}
                 isDownloading={isLoading}
                 fileName={getExpenseInvoiceFilename(collective, expense)}
+                describe="expense"
                 size={88}
-                showFileName
               />
             )}
           </ExpenseInvoiceDownloadHelper>
         </Box>
       )}
       {files.map((file, idx) => (
         <Box key={file.id || file.url} mr={3} mb={3}>
-          <UploadedFilePreview size={88} url={file.url} fileName={file.name} showFileName />
+          <UploadedFilePreview size={88} url={file.url} fileName={file.name} describe="attachment" />
```

Comment:
We're not yet matching the specs of the original issue:

> For the uploaded file, display `attached file` (if there are multiple we could just go attached file 1, attached file 2) - 'no filename' appears kind of broken and doesn't help the user understand what it is

As mentioned above, the `fileName` prop seems like the right place to implement this:

```es6
const getFileName = (intl, idx, totalNbFiles) => {
  if (totalNbFiles === 1) {
    return intl.formatMessage({ defaultMessage: 'Attached file' })
  } else {
    return intl.formatMessage({ defaultMessage: 'Attached file {number}' }, { number: idx + 1 })
  }
}
```

```suggestion
          <UploadedFilePreview size={88} url={file.url} fileName={getFileName(...)} />
```

Reply:
I decided to implement a version of this suggestion (see above). I kept the `fileName` in place and added a `description` prop where you pass a `FormattedMessage`, as is done for expense reports. For attached files, I implemented essentially what you suggested.

I think this now matches the specs of https://github.com/opencollective/opencollective/issues/5371.